Update to add info about syncing from a snapshot

pooja opened the pull request.

johnnymatthews commented on docs/build/start-building/deploy-your-application-to-the-filecoin-testnet.md:
```diff
@@ -5,12 +5,12 @@ description: We recommend all developers deploy their applications to the Fileco
 # Deploy your application to the Filecoin testnet
 
-If you have built your application and tested it on smaller-scale Filecoin networks (e.g. a [local network](https://docs.filecoin.io/build/start-building/interacting-with-the-network/#running-a-local-network) or the [Filecoin Nerpa network](https://docs.filecoin.io/build/start-building/interacting-with-the-network/#devnets)), you are now ready to deploy your application to the Filecoin testnet. W ...
+If you have built your application and tested it on smaller-scale Filecoin networks (e.g. a [local network](https://docs.filecoin.io/build/start-building/interacting-with-the-network/#running-a-local-network) or the [Filecoin Nerpa network](https://docs.filecoin.io/build/start-building/interacting-with-the-network/#devnets)), you are now ready to deploy your application to the Filecoin testnet. W ...
```
```suggestion
If you have built your application and tested it on smaller-scale Filecoin networks, like a [local network](https://docs.filecoin.io/build/start-building/interacting-with-the-network/#running-a-local-network) or the [Nerpa devnet](https://docs.filecoin.io/build/start-building/interacting-with-the-network/#devnets), then you are now ready to deploy your application to the Filecoin testnet! We currently recommend **all developers deploy their applications to the Filecoin testnet** in order to receive direct support from the Protocol Labs team and to be eligible for upcoming data storage rewards.
```

I have a personal vendetta against parentheses, mainly because they make translation a bit of a pain.

pooja replied:
Got it -- this makes the first sentence super long, so I'll fix another way (without introducing parens)

johnnymatthews commented on docs/build/start-building/deploy-your-application-to-the-filecoin-testnet.md:
```diff
@@ -5,12 +5,12 @@ description: We recommend all developers deploy their applications to the Fileco
 
 ::: tip
-If you are using a hosted service like [hosted Powergate](https://blog.textile.io/prepare-to-launch-expanding-free-access-to-filecoin-through-hosted-powergates/) or [Textile Buckets](https://docs.textile.io/buckets/), you don’t need to worry about any of this. Textile’s services are soon to be deployed against the Filecoin testnet, so if you are using hosted Textile products, your application wil ...
+**Recommended for most developers:** We recommend that most developers use a [hosted service like hosted Powergates or Textile Buckets](https://blog.textile.io/prepare-to-launch-expanding-free-access-to-filecoin-through-hosted-powergates/) to store your application data to the Filecoin testnet. These services are by far the fastest and friendliest way to get your application working with Filecoin ...
```
```suggestion
We recommend that **most developers use a hosted service** like hosted [Powergates or Textile Buckets](https://blog.textile.io/prepare-to-launch-expanding-free-access-to-filecoin-through-hosted-powergates/) to store your application data to the Filecoin testnet. These services are by far the fastest way to get your application working with Filecoin. If you are using a hosted Powergate, or Textile Buckets, you don’t need to worry about any of the information in this guide. Textile’s services are soon to be deployed against the Filecoin testnet, so if you are using hosted Textile products, your application will also be deployed on the Filecoin testnet.
```

johnnymatthews commented on docs/build/start-building/deploy-your-application-to-the-filecoin-testnet.md:
```diff
@@ -31,17 +33,47 @@ The high-level steps required to join the testnet are:
+
+| Snapshot Date | Download Link                                                                                                                                                                                                                        | Size (GiB) | sha256sum                                                        |
+| ------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ | ---------- | ---------------------------------------------------------------- |
+| 2020-09-05    | [https://very-temporary-spacerace-chain-snapshot.s3-us-west-2.amazonaws.com/Spacerace_2020-09-05_10-54-00-UTC.car](https://very-temporary-spacerace-chain-snapshot.s3-us-west-2.amazonaws.com/Spacerace_2020-09-05_10-54-00-UTC.car) | 12.6       | 8815c8b20c854323501fef52885cd2edf139c0a93e26918e2fd9a5478f5bdf24 |
```
This URL is funny.

johnnymatthews commented on docs/build/start-building/deploy-your-application-to-the-filecoin-testnet.md:
```diff
@@ -31,17 +33,47 @@ The high-level steps required to join the testnet are:
+### (7) Check your wallet balance
 
-If you are running into any issues joining the testnet, please join the [Filecoin Slack](http://filecoin.io/slack) and leave a message in the [#fil-storage-dev channel](https://app.slack.com/client/TEHTVS1L6/CRK2LKYHW). Our team will respond asap in the channel.
+Check your testnet wallet balance to make sure it contains >0 FIL.](https://docs.filecoin.io/how-to/join-a-network/#check-wallet-address-balance) You will not see any funds in your testnet wallet until your node is fully synced (which may take several days, depending on how you started chain sync -- if you are syncing from a snapshot, it should take <1 day for the balance to show up). Once you ha ...
```
```suggestion
Check your testnet wallet balance to make sure it contains more than 0 FIL.](https://docs.filecoin.io/how-to/join-a-network/#check-wallet-address-balance) You will not see any funds in your testnet wallet until your node is fully synced, which may take several days depending on how you started chain sync. If you are syncing from a snapshot, it should take less than 1 day for the balance to show up. Once you have more than 0 FIL in your testnet wallet, you can proceed to the next step.
```

johnnymatthews commented:
There's a few tiny changes I'd like to see. Also, I'd like to edit some of the formatting, but I'll do that after all the other edits have gone in. Adding formatting changes now will make Git freak out and think I'm rewriting everything.

ribasushi commented on docs/build/start-building/deploy-your-application-to-the-filecoin-testnet.md:
```diff
@@ -31,17 +33,47 @@ The high-level steps required to join the testnet are:
+
+### (3) Run the lotus daemon with the state snapshot
+
+Make sure you have a clean lotus install.
```
```suggestion
Make sure you have a clean lotus install, and that **you have backed up all your wallet keys!!!**
```

ribasushi commented on docs/build/start-building/deploy-your-application-to-the-filecoin-testnet.md:
````diff
@@ -31,17 +33,47 @@ The high-level steps required to join the testnet are:
+Then run the lotus daemon passing in the file name for the state snapshot you just downloaded.
+
+```
+lotus daemon --import-snapshot Spacerace_2020-09-05_10-54-00-UTC.car
````
```suggestion
lotus daemon --import-snapshot Spacerace_stateroots_snapshot_latest.car
```

ribasushi commented on docs/build/start-building/deploy-your-application-to-the-filecoin-testnet.md:
```diff
@@ -31,17 +33,47 @@ The high-level steps required to join the testnet are:
+
+### (4) Wait for the Filecoin chain to sync
+
+Because you are running the lotus daemon with a chain state snapshot, the chain sync process should be much faster than if you were syncing the chain from scratch. However, you will still need to sync the chain from the point at which the snapshot was generated to the current chain head. [You can run `lotus sync wait` to see the current sync status](https://docs.filecoin.io/how-to/join-a-network/ ...
```
```suggestion
Because you are running the lotus daemon from a known chain-state snapshot, the chain sync process will be significantly faster than if you were syncing the chain from scratch. However, you will still need to sync the chain from the point at which the snapshot was generated to the current chain head. [You can run `lotus sync wait` to see the current sync status](https://docs.filecoin.io/how-to/join-a-network/#chain-sync).
```

ribasushi reviewed (changes requested):
@pooja a number of corrections, sorry took a while

pooja commented on docs/build/start-building/deploy-your-application-to-the-filecoin-testnet.md:
```diff
@@ -31,17 +33,47 @@ The high-level steps required to join the testnet are:
-2. [**Run the lotus daemon.**](https://docs.filecoin.io/how-to/join-a-network/#get-started)
+### (2) Download the latest Filecoin chain state export
+
+Syncing the Filecoin chain from scratch can be a time-consuming process. Currently, chainsync takes about 1/4th of the lifetime of the network to complete. That is, a network 100 days old will take about 25 days to sync from scratch. Instead of syncing from scratch, we recommend that developers looking to deploy their applications quickly download a chain state export generated by the Protocol La ...
```
Yeah I think it's important to spell it out here so people know what to expect if they sync from scratch.

pooja commented:
Addressed all comments and going to merge so we can start linking people to the info! In the future, would be great to approve the PR anyways to let velocity remain high, instead of requesting changes. As an admin, I can merge regardless, but don't like blocking people with PR review esp with something low-risk like docs.
